paddle/fluid/platform/device_context.cc

```diff
@@ -441,10 +456,17 @@ void MKLDNNDeviceContext::SetBlob(const std::string& name,
   }
 
   // Find KeyBlob for current input shape
-  std::string cur_input_shape_str = platform::get_cur_input_shape_str();
   auto key_it = sBlob->find(cur_input_shape_str);
 
   if (key_it == sBlob->end()) {
+    // In cache clearing mode, cur_input_shape_cache_size defines max pblob
+    // capacity
+    if ((tid == kMKLDNNSessionID_CacheClearing) &&
+        (sBlob->size() == static_cast<size_t>(cur_input_shape_cache_size))) {
+      VLOG(2) << "tid=" << tid
+              << ", remove all head blob of shape: " << sBlob->begin()->first;
+      sBlob->erase(sBlob->begin()->first);
```

Comment:
why here remove sBlob->begin()->first? seems it is wrong,  should be erase(sBlob->begin()),  sBlob->begin()->first is string.
BTW, it doesn't really mean removing the head or first one because sBlob is std::unordered_map and its index method is not same with vector or queue,.

Reply:
map could erase by key, see: https://www.geeksforgeeks.org/map-erase-function-in-c-stl/

>  it doesn't really mean removing the head or first one because sBlob is 

Got it. I will change the LOG. For code, since we can erase any one of the sBlob, it runs successfully.

Reply:
you are right, there are 3 erase functions.